allo/ir/visitor.py

```diff
@@ -266,3 +289,24 @@ def visit_stmts(ctx, stmts):
     for stmt in stmts:
         results.append(visit_stmt(ctx, stmt))
     return results
+
+
+class ReplaceNames(ast.NodeTransformer):
+    def __init__(self, symbolic_mapping, var_map):
+        super().__init__()
+        self.symbolic_mapping = symbolic_mapping
+        self.var_map = var_map
+
+    def visit_Name(self, node):
+        if node.id in self.symbolic_mapping:
+            new_node = ast.parse(self.symbolic_mapping[node.id], mode="eval").body
+            return new_node
+        if node.id in self.var_map:
+            return ast.Constant(self.var_map[node.id])
+        return node
+
+
+def get_symbolic_expr(expr_node, mapping, var_map):
+    new_tree = ReplaceNames(mapping, var_map).visit(expr_node)
+    ast.fix_missing_locations(new_tree)
+    return ast.unparse(new_tree)
```

Comment:
Can you add comments to explain the function? What is it used for, and what does the return look like?